test/integration/error_log_test.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2021 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestLogfiles(t *testing.T) {
+	profile := UniqueProfileName("logfiles")
+	ctx, cancel := context.WithTimeout(context.Background(), Minutes(5))
+	defer Cleanup(t, profile, cancel)
+
+	tests := []struct {
+		command          string
+		args             []string
+		runCount         int // number of times to run command
+		expectedLogFiles int // number of logfiles expected after running command runCount times
+	}{
+		{
+			command:          "start",
+			args:             append([]string{"--memory=2200", "--wait=true"}, StartArgs()...),
+			runCount:         20, // calling this 20 times should create 2 files with 1 greater than 1M
```

Comment:
I'm concerned about running `minikube start` 20 times for this test -- this test will take minimum 20 minutes to run. 

Instead I would remove the `minikube start` subtest entirely and run `minikube status` 100 times (or whatever it takes) to make sure 2 log files are created.

Reply:
@priyawadhwa did a quick test and for around 500 times x `minikube status` only generates 87K, so we may need quite a lot of status calls (~5K) to fill up 1M and create 2 files.

Another alternative might be to fill up the log with some mock data(<1M) and expect that minikube handles the creation of new log correctly on subsequent calls. However not sure if that would be a best practice since this is an integration test.

Reply:
@bharathkkb Maybe try `minikube logs` much shorter running time that start and generates quite a few logs.

Reply:
@bharathkkb i really like that added integration test, I agree with @priyawadhwa lets try this test as a Sub Test for TestErrorSpam

we should not start a new cluster for this test, and it goes well with logging and spammming.
since we dont wanna spam TMPFolder

we can have a sub test that runs `minikube start --dry-run `

or minikube version